### ISSUE TITLE
[SR-13399] (Generic placeholder).Type broke for Any

### ISSUE THREAD

**Issue by swift-ci** [swiftlang/swift#55840]
|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-13399      |
   |Radar             | None         |
   |Original Reporter | Jessy (JIRA User)      |
   |Type              | Bug    |

   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 0         |
   |Component/s       |     |
   |Labels            | Bug        |
   |Assignee          | None      |
   |Priority          | Medium      |

   

   md5: f0784274714b253856da20d978d36950

  </details>

**Issue Description:**

The second and third print statements used to be the same. Now they are not. It causes me some problems.

```swift
func anyBug<Parameter>(_: Parameter.Type) {
  print(Parameter.self)
  print(Any.Type.self)
  print(Parameter.Type.self)
}

anyBug(Any.self)
```

**Comment by AnthonyLatsis** [comment]
This was fixed (probably by #41909).

**Comment by AnthonyLatsis** [comment]
Duplicate of #58367
